### PR TITLE
Make trace point-free

### PR DESCRIPTION
**Pull request by wafflepie**
Solves #23

### CODE REVIEW

File: packages/ramda-extension/src/trace.js [hunk 1]
```diff
@@ -14,6 +14,6 @@ import { curry } from 'ramda';
  *        compose(calculation2, R_.trace('Page A'), calculation1); // logs "Page A" and result of calculation1
  *
  */
-const trace = curry((x, y) => (console.log(x, y), y));
+const trace = curry(compose(last, tap(apply(console.log)), pair));
```
Comment (aizerin) [inline packages/ramda-extension/src/trace.js]: I would prefer more straightforward solution like

`useWith(tap, [curryN(2, console.log), identity])`

Comment (tommmyy) [reply packages/ramda-extension/src/trace.js]: I would prefer `binary` over `curryN`

Comment (aizerin) [reply packages/ramda-extension/src/trace.js]: That's even better :).

Comment (tommmyy) [reply packages/ramda-extension/src/trace.js]: actually, binary does not work in this case :-(

**Comment by wafflepie** [comment]
Thanks for the suggestion. However, I'm still trying to process why it works this way. :\

**Comment by aizerin** [comment]
@wafflepie what do you think exactly? I can explain it.

**Comment by wafflepie** [comment]
I thought that the return value would actually be a unary function (result of calling `curryN(2, console.log)` with the first argument).

**Comment by wafflepie** [comment]
Oh, never mind. The unary function gets passed as the first argument to `tap`, my bad. Got it now :)

**Comment by tommmyy** [comment]
There is another solution:

```
const trace = curry(compose(head, juxt([nthArg(1), console.log])));

// or

const trace = curry(compose(head, juxt([R_.lastArg, console.log])));
```

**Comment by tommmyy** [comment]
there is another one to consider.
```
const trace = useWith(R_.lastArg), [R_.log, R_.log]);
```

**Comment by wafflepie** [comment]
Although your last solution is probably the clearest in terms of implementation, it does not have the same behaviour, right? I think that console.log(foo, bar) is better than two console.logs, because it doesn't clutter the console as much and makes it more readable.

**Comment by jbradle** [comment]
close #23

**Pull request merged**